Finish Milestone 1

mistermaxis-zz opened the pull request:
### Completed Milestone 1:

- Added header section
- Added footer section
- Styled and positioned elements using bootstrap

mistermaxis-zz committed:
Install and setup linter

mistermaxis-zz committed:
Add images and icons for header

mistermaxis-zz committed:
Add elements for banner part of header

mistermaxis-zz committed:
Add style.css with dummy class

mistermaxis-zz committed:
Add elements for nav part of header.

mistermaxis-zz committed:
Add bootstrap CDN

mistermaxis-zz committed:
Style header and fix linter errors

mistermaxis-zz committed:
Change to header element

mistermaxis-zz committed:
Delete package-lock.json

mistermaxis-zz committed:
Merge branch 'footer' into dev-milestone-1

mistermaxis-zz committed:
Merge branch 'master' into dev-milestone-1

mistermaxis-zz committed:
Deleted hint report html file

mistermaxis-zz committed:
Remove section from header and change date to span

mistermaxis-zz committed:
remove link to stylesheet.css

mistermaxis-zz committed:
Add hint-report to .gitignore

mistermaxis-zz committed:
Add style for footer in style.css

jubaan reviewed (changes requested):
[![Microverse][microverse-shield]][microverse-url]
<div>
<h1 align="center"> ⚠️ 🛠️ ⚠️ 🛠️ Changes Required 🛠️ ⚠️ 🛠️ ⚠️ </h1>
<br />
<p><b>"Fail fast, fail often"</b> and don't let this discourage you, because <b>only by failing do we learn to succeed</b>.</p>
<details open="open">
  <summary>Review Summary</summary>
  <ol>
    <li><a href="#congrats"> 💯 Awesome work 👏🏼 👏🏼 </a></li>
    <li><a href="#mandatory"> ⚠️ Things to Improve </a> 🚨 🚨 </li>
    <li><a href="#optional"> 💡Nice to have and other comments ✍🏼 ✍🏼 </a></li>
    <li><a href="#contact"> 🧔🏼 About your Reviewer </a> (you don't know it... but we are just automated bots 🤖) </li>
    <li><a href="#help"> 🆘 <b> 😭 Feeling stuck? or Need help</b>? 🙏🏼 🙏🏼 </a></li>
  </ol>
</details>
</div>

<h3 id="congrats"> 1. :100: Awesome work</h3>
<details close="close">
<summary> [click to open] Big or little achievements, but achievements at the end. <b>Feel proud that you're learning and improving everyday.</b></summary>
<!-- BEGIN Awesome work -->

<!-- END Awesome work  -->
</details>

<hr>
<h3 id="mandatory"> 2. ⚠️ Things to Improve (or the To-Do list to get the project approved) </h3> 
<details close="close">
<summary> [click to open] Working on these observations will help you advance and get closer to the goal 🎯, postpone them or do them half-ways and they'll haunt you 👻 👻... FOREVER!! (It will feel like that, trust me). </summary>
<!-- BEGIN Review -->

- [ ] The website is taking the whole width of the browser while the original desing has a max-width applied. Please review this and make the corresponding changes.
![image](https://user-images.githubusercontent.com/25774518/110414786-38989a80-8056-11eb-804b-07bb2d403f2c.png)
- [ ] On desktop version there is no login icon
![image](https://user-images.githubusercontent.com/25774518/110414863-6a116600-8056-11eb-85d7-92646128f8dc.png)
- [ ] In tablet view applies the same for the icon and also the Newsweek logo has a margin left
![image](https://user-images.githubusercontent.com/25774518/110414996-a2b13f80-8056-11eb-9254-fc12cca3e696.png)
- [ ] Footer does not match the layout nor style of the original design in all of the views.
![image](https://user-images.githubusercontent.com/25774518/110415070-d2f8de00-8056-11eb-951e-a66f6a7fa7cb.png)
![image](https://user-images.githubusercontent.com/25774518/110415113-e86e0800-8056-11eb-8b59-94d43017b38f.png)
- [ ] `href` attribute should have `#` if is not redirecting to any website.
- [ ] `alt` attribute should not be empty.
- [ ] The search should be an input
- [ ] The social icons should be links
- [ ] The menu items of the footer all should be links also.
- [ ] Review the CSS stylesheet and remove all CSS rules that have bootstrap counterpart. Add the bootstrap class to the elements that need it.


<!-- END Review -->
</details>
<p><b>NOTE</b>This list is or could be divided in what is show in this comment and inline comments posted below.<p>
<hr>

<h3 id="optional"> 3. 💡 Nice to have and other comments </h3>
<details close="close">
<summary> [click to open] </h3></summary>
<!-- BEGIN Optional and Comments -->

- [x] Nothing more to add.

<!-- END Optional and Comments -->
</details>

<h3 id="contact"> 4. 🧔🏼 About your Reviewer </h3>
<details close="close">
<summary> [click to open] Say "hello world!" on Slack: @jubaan</summary>
<br />
<div align="center">
  <a src="https://www.github.com/jubaan">
    <img src="https://avatars.githubusercontent.com/u/25774518?s=400&u=ae20ddf2d6fb3cae9e2f4044afe3b93d19e5b69e&v=4" alt="Julio Añoveros" title="@jubaan" width="100px" heigh="100px"/>
  </a>
  <p>
    Julio Añoveros | @jubaan</br></br>
    <b>Full-stack Web Developer</b> and</br>
    🤖 Code Reviewer at <a href="https://www.microverse.org/?grsf=kfm2ej">Microverse</a>:rocket:
  </p>
</div>
</details>

<h3 id="help"> 5. 🆘 <b> 😭 Feeling stuck? or Need help</b>? 🙏🏼 🙏🏼</h3>
<details close="close">
  <summary> [click to open] <b>Feel stuck?</b> </summary>
  <br>
  <p> First breath... just do it. Take 15-20 min away from the computer, stretch, meditate, do some exercise, drink water, or do whatever is necessary to distract your mind from the obstacle.</p>
  <p><b>Sometimes you just need to get away to see things with a new perspective.</b></p>
</details>

<details close="close">
  <summary> [click to open] <b>Need help?</b> </summary>
  <br>
  <p> Send me a DM in <a href="https://www.microverse.org/?grsf=kfm2ej">Microverse</a> Students Slack, you'll find me as @jubaan. If I'm not available there are other Code Reviewers you can contact. They'll help as well.</p>
  <p>Remember that <a href="https://www.microverse.org/?grsf=kfm2ej">Microverse</a> has other channels where you can get an answer to your questions:</p>
  <ol>
    <li>The <b>students channel 🧑‍🎓 </b> in slack,</li>
    <li>your <b>teammates 👥 </b>,</li>
    <li>your <b>mentor 👨🏼 👩🏼 </b>, and</li>
    <li><b>Student Success 🏫 </b>.</li>
  </ol>
  <p><b> DON'T hesitate to ask! </b></p>
</details>
<p>If all of the above doesn't help... get a <a href="https://rubberduckdebugging.com/">rubber duck</a> <img src="https://www.pinclipart.com/picdir/big/130-1305924_rubber-ducky-clipart-eastern-iowa-duck-race-duck.png" width="18px" height="18px">... it would be great to own one 😄. </p>

<h2 align="center"> Keep up the great work :muscle: and Happy Coding :smiley: :computer:!! </h1>

[microverse-shield]: https://img.shields.io/badge/-microverse-%236F23FF?style=for-the-badge
[microverse-url]: https://www.microverse.org/?grsf=kfm2ej
[bootstrap-shield]: https://img.shields.io/badge/Bootstrap-white?style=for-the-badge&logo=bootstrap
[bootstrap-url]: https://www.microverse.org/?grsf=kfm2ej

mistermaxis-zz committed:
Add input for search

mistermaxis-zz committed:
Add "alt" value to img elements

mistermaxis-zz committed:
Add '#' value to href property in <a> tags

mistermaxis-zz committed:
Update index.html

Add <a> tags to social media icons and menu items on footer

mistermaxis-zz committed:
Update index.html

- Change login button and link so they switch when the screen changes.
. Add left margin to header icon when positioned on the left for small screens.

mistermaxis-zz committed:
Update index.html and style.css

- Change classes in elements to use bootstrap instead.
- Change footer elements to be properly position in smaller screens.

mistermaxis-zz committed:
Update index.html

Group footer links to behave as in the original website and escaped & symbol in footer element.

mistermaxis commented:
### Changes implemented:

#### General:
- Removed standard CSS classes where logical to use bootstrap instead.

#### Header:
- Changed container properties to not extend 100% of the viewport.
- Use bootstrap to switch a login icon with a login link on the desktop view.
- Applied a left margin to the logo for tablet and mobile views.
- Added input in the "search" section.
- Added '#' to <href> elements.
- Added description to "alt" properties.

#### Footer:
- Changed footer layout to match the original in all views.
- Made social media icons into links.
- Made footer items into links.

yokoworks commented on index.html:
```diff
@@ -1,12 +1,107 @@
+      <div class="row2 d-flex flex-column ">   
+        <div class="d-flex flex-wrap align-content-start">
+            <i class="font-weight-bold border border-0">Editions:</i>
+            <a href="#"><i>U.S. Edition</i></a>
+            <a href="#"><i>日本</i></a>
+            <a href="#"><i>Pakistan</i></a>
+            <a href="#"><i>Polska</i></a>
+            <a href="#"><i>România</i></a>
```
Usually, `<i>` tag is used only when there is not a more appropriate semantic tag, or to display an icon, or [to indicate a technical term](https://www.w3schools.com/tags/tag_i.asp).

In this case, instead of `<i>` tag, it should be simply `<a href="#">U.S. Edition</a>` or you can also use list tag for the whole links like `<ul><li><a href="#">U.S. Edition</a></li><li><a href="#">日本</a></li><li><a href="#">Pakistan</a></li></ul>`. 

You can see how the original website is coded with your [inspector in your Google Chrome browser](https://developers.google.com/web/tools/chrome-devtools/dom) 👍🏼

yokoworks commented on style.css:
```diff
@@ -0,0 +1,151 @@
+  border-radius: 24px;
+  font-weight: bolder;
+  font-size: 12px;
+  padding: 0.1rem 0.5rem 0.2rem 0.6rem;
```
Now they are specified in the CSS file, but in this project, for our learning, we are supposed to specify these kinds of styles with Bootstrap **everywhere** as possible as we can. 
Could you use Bootstrap instead of writing them in the CSS file? 🚀

yocosaka commented on index.html:
```diff
@@ -1,12 +1,107 @@
+
+    <footer class="footer">
+      <div class="row1 d-flex flex-column flex-lg-row">
+        <svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" viewBox="0 0 395.6 60" style="enable-background: new 0 0 395.6 60;" xml:space="preserve"><g><path class="st0" d="M58.7,10.8h-5.6v48.3H42.8l-25.1-33v22.1h5.6v10.8H0V48.3h5.6V10.8H0V0h17.6L41,30.7V10.8h-5.6V0h23.2	L58.7,10.8L58.7,10.8z M95.8,38c0,0.9,0,1.8-0.1,2.7H70 ...
```
Would you add <a> tag to the logo in the footer?
The Footer logo is also supposed to be a link as well as one in the header 👍🏼

yokoworks reviewed (changes requested):
##  Status: Changes Required♻️ 

**Great job so far, @mistermaxis**👍  

You have made an amazing job🚀
There are some points to improve before you go ahead.
Let's cover a few details together to make your project better😊 

### Required changes: 
- [x] Add a link (`<a>` tag) to the footer-logo
- [x] Use Bootstrap CSS classes everywhere for styling
- [x] Remove extra `<i>` tag in the footer links
- [x] Remove extra padding on both sides of the width in the mobile and tablet
- [x] Remove extra separate style `|`  in footer links in the mobile

Please request another review once you are ready👍

### With all my respect 🍀

Yoko Saka

mistermaxis-zz committed:
Update index.html and  style.css

- Change svg element to img element.
- Remove i tags from footer links.
- Remove padding from header.
- Change css classes to bootstrap classes.

mistermaxis-zz committed:
Change rem to px in logo img element

mistermaxis-zz committed:
Remove px from logo img element

mistermaxis-zz committed:
Fix linter errors

mistermaxis commented:
## Changes Implemented:

### General:
- Change CSS classes for bootstrap classes.

### Header:
- Removed padding from the header section.

### Footer:
- Added a link to the footer logo.
- Removed the 'i' tags from the footer links.
- Removed border for the footer links in smaller screens.

wathigo reviewed (approved):
## Approved :heavy_check_mark:  
Great job implementing the suggested changes.

**You may proceed to the next milestone.**

**Happy Coding :computer: 
Simon Wathigo**

mistermaxis commented:
The milestone has been approved, but for some reason, in the dashboard, it appears as if I hadn't submitted anything so I'm submitting it again. Thank you.
